lib/MySQL_Monitor.cpp

```diff
@@ -3353,6 +3353,31 @@ void MySQL_Monitor::process_discovered_topology(const std::string& originating_s
 	}
 }
 
+/**
+* @brief Check if a list of servers is matching the description of an AWS RDS Multi-AZ DB Cluster.
+* @details This method takes a vector of discovered servers and checks that there are exactly three which are named "instance-[1|2|3]" respectively, as expected on an AWS RDS Multi-AZ DB Cluster.
+* @param discovered_servers A vector of servers discovered when querying the cluster's topology.
+* @return Returns 'true' if all conditions are met and 'false' otherwise.
+*/
+bool MySQL_Monitor::is_aws_rds_multi_az_db_cluster_topology(const std::vector<MYSQL_ROW>& discovered_servers) {
+	if (discovered_servers.size() != 3) {
+		return false;
+	}
+
+	const std::vector<std::string> instance_names = {"-instance-1", "-instance-2", "-instance-3"};
+	int identified_hosts = 0;
+	for (std::string instance_str : instance_names) {
+		for (MYSQL_ROW server : discovered_servers) {
+			std::string current_discovered_hostname = server[2];
```

Comment:
Minor thing too, here we are assuming that `server[2]` can't be `NULL`. This is already done in the original implementation, so it's not a big deal, and without the table definition not possible to tell, but would be nice to know if this can be assumed. If not, this might need revisiting if the feature changes/expands.

Reply:
Yes, it seems to be assumed in the previous implementation. But you are right, I can add a null-check.